### DISCUSSION TITLE
Avalonia Mobile, open pages

### DISCUSSION THREAD

**Discussion by bsdRuf** [AvaloniaUI/Avalonia#15652] [Q&A] [answered]
Hello, I've seen the tutorial page to [navigate views](https://docs.avaloniaui.net/docs/tutorials/todo-list-app/navigate-views).

Just to be sure. I would need a declared variable of every ViewModel I may need? To access the functions that are inside because the `ViewModelBase _contentViewModel` cant?

Do we have a copy pastable (from a example or sth) for the back button? What do you think would be the best approach to make the back button work? Have a history list of the viewmodels called and overrite the back button?

Since the desktop has Show and ShowDialog I feel like on the mobile side something is missing.

Thanks for all the samples, they helped out quite a few times.

**Comment by thevortexcloud** [comment, marked as answer]
> I would need a declared variable of every ViewModel I may need?

Depends on how you implement it. You can do it that way, but it gets fairly big and unwieldy. For larger apps you typically use a router. ReactiveUI for example has one built in that supports history:

https://docs.avaloniaui.net/docs/concepts/reactiveui/routing

Otherwise it's not that hard to make your own.

> Since the desktop has Show and ShowDialog I feel like on the mobile side something is missing.

You need to implement dialogues yourself for SPA. You can use helper libraries like DialogHost though.

https://github.com/AvaloniaUtils/DialogHost.Avalonia

You can also use other libraries built around it, like the material theme, which will also give you something that looks like a native android app.

https://github.com/AvaloniaCommunity/Material.Avalonia

**Reply by bsdRuf** [reply]
Its not working, can you tell me whats the mistake(s) in my example?

[Github repo](https://github.com/bsdRuf/RoutingExample)

**Reply by thevortexcloud** [reply]
Your link does not work. Is the repo private?

**Comment by timunie** [comment]
todolist is desktop only for the moment. It uses windowing system, which mobile doesn't have. 

> [!NOTE]
> This tutorial was done when we didn't have a mobile backend.
> We are working on a modernization: https://github.com/AvaloniaUI/Avalonia.Samples/pull/76 . Your feedback is welcome if you have any.
